<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Robin Wagner (UoM Laptop)" />

<meta name="date" content="2023-08-23" />

<title>CRLM_Subtyping</title>

<script src="site_libs/header-attrs-2.24/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">NK_CRLM</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/robin-a-wagner/NK_CRLM">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">CRLM_Subtyping</h1>
<h4 class="author">Robin Wagner (UoM Laptop)</h4>
<h4 class="date">2023-08-23</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-08-23
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>NK_CRLM/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230823code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230823)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230823code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230823)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomrobinawagnerNKCRLMtree107c71f618a1b2056ad46b2a9ca9f9ec82261daetargetblank107c71fa">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/robin-a-wagner/NK_CRLM/tree/107c71f618a1b2056ad46b2a9ca9f9ec82261dae" target="_blank">107c71f</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomrobinawagnerNKCRLMtree107c71f618a1b2056ad46b2a9ca9f9ec82261daetargetblank107c71fa"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/robin-a-wagner/NK_CRLM/tree/107c71f618a1b2056ad46b2a9ca9f9ec82261dae" target="_blank">107c71f</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store

Untracked files:
    Untracked:  data/20230822_Shembrey_NK_Signature.csv
    Untracked:  data/DGEList_3prime_Org_171samples_rm4Tum_9Untreated.RDS
    Untracked:  data/DGEList_Filtered_TCGA_COAD_ForCarolyn.RDS
    Untracked:  data/Genomics_WES_WGS_ProteinCoding_ImpactHighModerate_Canonicals_UniqueGenes.txt
    Untracked:  data/Hollande_DGEList_Tum_Raw.RDS
    Untracked:  data/iCMS_template.csv
    Untracked:  output/20230823_Shembrey_NK_Score_Hollande_CRLM.csv

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/CRLM_Subtyping.Rmd</code>) and
HTML (<code>docs/CRLM_Subtyping.html</code>) files. If you’ve configured
a remote Git repository (see <code>?wflow_git_remote</code>), click on
the hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/robin-a-wagner/NK_CRLM/blob/107c71f618a1b2056ad46b2a9ca9f9ec82261dae/analysis/CRLM_Subtyping.Rmd" target="_blank">107c71f</a>
</td>
<td>
Robin Wagner (UoM Laptop)
</td>
<td>
2023-08-23
</td>
<td>
Amended boxplot colors
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/robin-a-wagner/NK_CRLM/7d172e2837c573d728c68935e723e46d1dfba4bf/docs/CRLM_Subtyping.html" target="_blank">7d172e2</a>
</td>
<td>
Robin Wagner (UoM Laptop)
</td>
<td>
2023-08-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/robin-a-wagner/NK_CRLM/blob/afb3d73e1b216e12c2f8cf6896969eb38223b5a6/analysis/CRLM_Subtyping.Rmd" target="_blank">afb3d73</a>
</td>
<td>
Robin Wagner (UoM Laptop)
</td>
<td>
2023-08-23
</td>
<td>
Silenced messages on CMScaller
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/robin-a-wagner/NK_CRLM/8988bffe11720b72c7f357c9fcc8967d3823f834/docs/CRLM_Subtyping.html" target="_blank">8988bff</a>
</td>
<td>
Robin Wagner (UoM Laptop)
</td>
<td>
2023-08-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/robin-a-wagner/NK_CRLM/blob/f4b343d9a0656de03dabe92c0b77d95c61e3cca7/analysis/CRLM_Subtyping.Rmd" target="_blank">f4b343d</a>
</td>
<td>
Robin Wagner (UoM Laptop)
</td>
<td>
2023-08-23
</td>
<td>
Comitting my first-pass draft
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/robin-a-wagner/NK_CRLM/d5a0814c09087341f240b01b612f3d260e194c46/docs/CRLM_Subtyping.html" target="_blank">d5a0814</a>
</td>
<td>
Robin Wagner (UoM Laptop)
</td>
<td>
2023-08-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/robin-a-wagner/NK_CRLM/blob/dcb7ccbd9d92b021ffd0d2968a5e4b72802bb2ef/analysis/CRLM_Subtyping.Rmd" target="_blank">dcb7ccb</a>
</td>
<td>
Robin Wagner (UoM Laptop)
</td>
<td>
2023-08-23
</td>
<td>
wflow_publish("analysis/CRLM_Subtyping.Rmd")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>In this code I will use nearest template prediction to classify the
consensus molecular subtype (CMS) and intrinsic CMS of each CRLM sample
using the CMScaller package.</p>
<pre class="r"><code>library(dplyr)
library(CMScaller)
library(ggplot2)
library(RColorBrewer)</code></pre>
<div id="prepare-data" class="section level3">
<h3>Prepare data</h3>
<p>Read in and prepare the raw counts data for the 58 samples.</p>
<pre class="r"><code>expr &lt;- readRDS(&quot;data/Hollande_DGEList_Tum_Raw.RDS&quot;); dim(expr)</code></pre>
<pre><code>Loading required package: edgeR</code></pre>
<pre><code>Loading required package: limma</code></pre>
<pre><code>[1] 25702    58</code></pre>
<p>Remove duplicated sample.</p>
<pre class="r"><code>expr$samples &lt;- expr$samples[-12,]
expr$counts &lt;- expr$counts[,-12]
dim(expr$samples); dim(expr$counts)</code></pre>
<pre><code>[1] 57 43</code></pre>
<pre><code>[1] 25702    57</code></pre>
<pre class="r"><code>all(colnames(expr$counts) == rownames(expr$samples))</code></pre>
<pre><code>[1] TRUE</code></pre>
<p>Filter out primary tumours as these are irrelevant to our
analysis.</p>
<pre class="r"><code>expr$samples &lt;- expr$samples %&gt;% filter(!Tissue == &quot;CoT&quot;); summary (expr$samples$Tissue)</code></pre>
<pre><code>CoT  LT 
  0  44 </code></pre>
<pre class="r"><code>expr$counts &lt;- expr$counts[,rownames(expr$samples)]; dim(expr$counts)</code></pre>
<pre><code>[1] 25702    44</code></pre>
<p>Filter out lowly-expressed genes.</p>
<pre class="r"><code>keep &lt;- filterByExpr(expr$counts)</code></pre>
<pre><code>No group or design set. Assuming all samples belong to one group.</code></pre>
<pre class="r"><code>expr_filt &lt;- expr[keep, ]; dim(expr_filt)</code></pre>
<pre><code>[1] 9888   44</code></pre>
<p>Normalise for library size.</p>
<pre class="r"><code>expr_norm &lt;- calcNormFactors(expr_filt, method = &quot;TMM&quot;)</code></pre>
</div>
<div id="perform-subtyping" class="section level3">
<h3>Perform subtyping</h3>
<p>Perform CMS classification using inbuilt template.</p>
<pre class="r"><code>CMS &lt;- CMScaller(emat = expr_norm$counts, RNAseq=TRUE, FDR=0.05); head(CMS)</code></pre>
<p><img src="figure/CRLM_Subtyping.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/robin-a-wagner/NK_CRLM/blob/8988bffe11720b72c7f357c9fcc8967d3823f834/docs/figure/CRLM_Subtyping.Rmd/unnamed-chunk-7-1.png" target="_blank">8988bff</a>
</td>
<td>
Robin Wagner (UoM Laptop)
</td>
<td>
2023-08-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
CMS1 CMS2 CMS3 CMS4 &lt;NA&gt; 
   4    4    3   19   14 </code></pre>
<pre><code>            prediction    d.CMS1    d.CMS2    d.CMS3    d.CMS4    p.value
JV388_LT          CMS3 0.6818851 0.6751093 0.6464738 0.8253085 0.02697303
KG078_LT          &lt;NA&gt; 0.7375867 0.6906808 0.7137610 0.8315222 0.89410589
PCC002_LT1a       &lt;NA&gt; 0.8140515 0.6483671 0.7308334 0.7996499 0.03596404
PCC002_LT1b       CMS4 0.7209306 0.6957662 0.7393632 0.5656365 0.00100000
PDO032_LT         CMS4 0.5960818 0.7169246 0.6856722 0.5786111 0.00100000
PDW486_LT         CMS4 0.6053514 0.7252045 0.6907481 0.5934999 0.00100000
                    FDR
JV388_LT    0.039560440
KG078_LT    0.972144135
PCC002_LT1a 0.051045728
PCC002_LT1b 0.001692308
PDO032_LT   0.001692308
PDW486_LT   0.001692308</code></pre>
<pre class="r"><code>summary(CMS$prediction)</code></pre>
<pre><code>CMS1 CMS2 CMS3 CMS4 NA&#39;s 
   4    4    3   19   14 </code></pre>
<p>Create templates for iCMS. Here genes up in iCMS2 and down iCMS3 are
designated iCMS2 genes. The reverse is true for iCMS3.</p>
<pre class="r"><code>iCMS_template &lt;- read.csv(&quot;data/iCMS_template.csv&quot;); head(iCMS_template)</code></pre>
<pre><code>         symbol class
1        CXCL14 iCMS2
2 LY6G6F-LY6G6D iCMS2
3          EREG iCMS2
4         TIMP3 iCMS2
5         KRT23 iCMS2
6        RUBCNL iCMS2</code></pre>
<pre class="r"><code>iCMS_template$probe &lt;- fromTo(key = iCMS_template$symbol, id.in = &quot;symbol&quot;, id.out = &quot;entrez&quot;)</code></pre>
<pre><code>no corresponding entrez for symbol key(s) LY6G6F-LY6G6D; RAB5IF; RTF2; ATP5F1E; SINHCAF; AC004130.1; SMIM26; RETREG1; AL445524.1; DEPP1; AC020656.1; ATP5F1A; TENT5A; AP001816.1; DGLUCY; AC005833.1; SMIM31</code></pre>
<pre><code>Warning: 17 identifiers not translated, NA&#39;s returned</code></pre>
<p>Classify according to iCMS.</p>
<pre class="r"><code>iCMS &lt;- CMScaller(expr_norm$counts, templates = iCMS_template, RNAseq = T, FDR = 0.05); head(iCMS)</code></pre>
<p><img src="figure/CRLM_Subtyping.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/robin-a-wagner/NK_CRLM/blob/8988bffe11720b72c7f357c9fcc8967d3823f834/docs/figure/CRLM_Subtyping.Rmd/unnamed-chunk-9-1.png" target="_blank">8988bff</a>
</td>
<td>
Robin Wagner (UoM Laptop)
</td>
<td>
2023-08-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre><code>
iCMS2 iCMS3  &lt;NA&gt; 
   18    15    11 </code></pre>
<pre><code>            prediction   d.iCMS2   d.iCMS3     p.value       FDR
JV388_LT          &lt;NA&gt; 0.6857759 0.7278128 0.150849151 0.1843712
KG078_LT          &lt;NA&gt; 0.6904140 0.7234145 0.281718282 0.3178360
PCC002_LT1a      iCMS2 0.5960954 0.8029136 0.001000000 0.0017600
PCC002_LT1b      iCMS2 0.6611603 0.7502447 0.005994006 0.0101437
PDO032_LT         &lt;NA&gt; 0.7040763 0.7101243 0.856143856 0.8760542
PDW486_LT        iCMS3 0.7684975 0.6398528 0.001000000 0.0017600</code></pre>
<pre class="r"><code>summary(iCMS$prediction)</code></pre>
<pre><code>iCMS2 iCMS3  NA&#39;s 
   18    15    11 </code></pre>
<p>Load NK score data.</p>
<pre class="r"><code>nk_score &lt;- read.csv(&quot;output/20230823_Shembrey_NK_Score_Hollande_CRLM.csv&quot;, row.names = 1); head(nk_score)</code></pre>
<pre><code>       sample      score quartile  designation
1    JV388_LT -0.4628994        1          low
2    KG078_LT -0.4401634        3 intermediate
3 PCC002_LT1a -0.4714542        1          low
4 PCC002_LT1b -0.4519359        3 intermediate
5   PDO032_LT -0.4574141        2 intermediate
6   PDW486_LT -0.4471963        3 intermediate</code></pre>
<p>Visualise the association between subtypes and NK score.</p>
<pre class="r"><code>plot_data &lt;- data.frame(sample = colnames(expr_norm$counts),
                        CMS = CMS$prediction,
                        iCMS = iCMS$prediction,
                        nk_score = nk_score$score) %&gt;%
  tidyr::gather(key = system, 
         value = subtype,
         c(&quot;CMS&quot;, &quot;iCMS&quot;)); head(plot_data)</code></pre>
<pre><code>Warning: attributes are not identical across measure variables; they will be
dropped</code></pre>
<pre><code>       sample   nk_score system subtype
1    JV388_LT -0.4628994    CMS    CMS3
2    KG078_LT -0.4401634    CMS    &lt;NA&gt;
3 PCC002_LT1a -0.4714542    CMS    &lt;NA&gt;
4 PCC002_LT1b -0.4519359    CMS    CMS4
5   PDO032_LT -0.4574141    CMS    CMS4
6   PDW486_LT -0.4471963    CMS    CMS4</code></pre>
<pre class="r"><code>ggplot(plot_data, aes(x = subtype, y = nk_score, fill = subtype)) + 
  geom_boxplot() + 
  geom_point(position = position_dodge(0.9), color = &quot;gray20&quot;) + 
  facet_wrap(~system, scales = &quot;free_x&quot;) +
  theme_bw() +
  theme(text = element_text(size = 15)) +
  scale_fill_manual(values = brewer.pal(8, &quot;Paired&quot;))</code></pre>
<p><img src="figure/CRLM_Subtyping.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-1">
Past versions of unnamed-chunk-11-1.png
</button>
</p>
<div id="fig-unnamed-chunk-11-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/robin-a-wagner/NK_CRLM/blob/8988bffe11720b72c7f357c9fcc8967d3823f834/docs/figure/CRLM_Subtyping.Rmd/unnamed-chunk-11-1.png" target="_blank">8988bff</a>
</td>
<td>
Robin Wagner (UoM Laptop)
</td>
<td>
2023-08-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.1 (2023-06-16)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Ventura 13.3

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Australia/Melbourne
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] edgeR_3.42.4       limma_3.56.2       RColorBrewer_1.1-3 ggplot2_3.4.3     
[5] CMScaller_0.99.2   dplyr_1.1.2        workflowr_1.7.0   

loaded via a namespace (and not attached):
 [1] tidyr_1.3.0       sass_0.4.7        utf8_1.2.3        generics_0.1.3   
 [5] lattice_0.21-8    stringi_1.7.12    digest_0.6.33     magrittr_2.0.3   
 [9] evaluate_0.21     grid_4.3.1        fastmap_1.1.1     rprojroot_2.0.3  
[13] jsonlite_1.8.7    processx_3.8.2    whisker_0.4.1     ps_1.7.5         
[17] promises_1.2.1    httr_1.4.7        purrr_1.0.2       fansi_1.0.4      
[21] scales_1.2.1      jquerylib_0.1.4   cli_3.6.1         rlang_1.1.1      
[25] munsell_0.5.0     withr_2.5.0       cachem_1.0.8      yaml_2.3.7       
[29] tools_4.3.1       colorspace_2.1-0  locfit_1.5-9.8    httpuv_1.6.11    
[33] vctrs_0.6.3       R6_2.5.1          lifecycle_1.0.3   git2r_0.32.0     
[37] stringr_1.5.0     fs_1.6.3          pkgconfig_2.0.3   callr_3.7.3      
[41] pillar_1.9.0      bslib_0.5.1       later_1.3.1       gtable_0.3.3     
[45] glue_1.6.2        Rcpp_1.0.11       highr_0.10        xfun_0.40        
[49] tibble_3.2.1      tidyselect_1.2.0  rstudioapi_0.15.0 knitr_1.43       
[53] farver_2.1.1      htmltools_0.5.6   labeling_0.4.2    rmarkdown_2.24   
[57] compiler_4.3.1    getPass_0.2-2    </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
